apps/src/lab2/levelEditors/levelData/EditMusicLevelData.tsx

```diff
@@ -32,10 +33,16 @@ interface EditMusicLevelDataProps {
 const EditMusicLevelData: React.FunctionComponent<EditMusicLevelDataProps> = ({
   initialLevelData,
 }) => {
+  const blockMode =
+    (
+      useAppSelector(
+        state => state.lab.levelProperties?.levelData
```

Comment:
Good points!

 I fixed for your first point so that we're using `initialLevelData` as the source of truth. This works as expected, or at least it works in that we only call the `setUpBlocklyForMusicLab` function once, resulting in the wrong function blocks if you switch modes. This actually how it works on levelbuilder (ie. staging) now too, which isn't ideal.

This morning I was planning to add some custom procedure block definitions for advanced mode in another branch. Now I think I'll do that based off this branch so it can be reviewed separately. I think I have a way to get the blocks we want registered without breaking things. Given that this branch maintains parity with staging, would it be fair to consider that a follow-up task?